Makefile

```diff
@@ -81,11 +81,8 @@ PKG_TEST_DIRS := $(filter-out $(INTEG_TEST_ROOT)%,$(TEST_DIRS))
 #   Packages are specified as import paths.
 GOCOVERPKG_ARG := -coverpkg="$(PROJECT_ROOT)/common/...,$(PROJECT_ROOT)/service/...,$(PROJECT_ROOT)/client/...,$(PROJECT_ROOT)/tools/..."
 
-dep-ensured:
-	./install-dep.sh
-	dep ensure
-
 yarpc-install:
+	go mod vendor
```

Comment:
I still think using [gobin](https://github.com/myitcv/gobin) would be better here. In the same issue that you linked eariler (golang/go#32504) it seems to me that some tooling will be the final solution eventually, not `tools.go`.

Reply:
for this issue, I need some opinion from my colleagues @shreyassrivatsan @samarabbas @dmetzgar, any idea?

FYI, this PR will not be landed until I get back from, roughly 1 week later.
In the meantime, I will try to do some investigation

Reply:
By using gobin, those two tools won't be added to `go.mod` which means less dependencies. Also, you don't need `go mod vendor`.

Reply:
> By using gobin, those two tools won't be added to `go.mod` which means less dependencies. Also, you don't need `go mod vendor`.

for our use case, we prefer to use the binary from dependency `go.uber.org/thriftrw` and `go.uber.org/yarpc` (we are already depending on these 2 lib).
assuming that my understanding of gobin is correct ([ref](https://github.com/myitcv/gobin#examples-using--m)), tool dependency still needs to be declared.

so (still assuming that my understanding is correct) the difference is basically whether it is a good idea to include another dependency.

Reply:
Ah, okay. If they are library dependencies as well, I guess this solution is ok.